paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -253,6 +255,11 @@ class MKLDNNHandlerT {
         std::static_pointer_cast<dnnl::memory>(dev_ctx_.GetBlob(target_key));
 
     if (target_memory_p == nullptr) {
+      if (custom_func) {
```

Comment:
Yes, it is intentionally. *Custom reorder function* is used to set an appropriate reorder for `data_format`. However, `user_md! = target_md` may differ in data type. In the case of bf16, the original weights (user_md) are float, while for calculations (target_md) we need them in bf16 and it is converted in this reorder.

Reply:
Ok, then if the _additional_ thing you want to do is data type conversion I'd suggest to create a helper function for this task explicitly named anything close to `ConvertMemDataType`. With it there would be clear intention of control-flow logic: 1st reorder data, 2nd convert memory data type. This `AcquireMemoryWithReorder` function is already very complicated, very hard to understand it's control-flow and I suppose hard to debug. Not mentioning it's maintenance and testing. IMHO this function is doing to much things.

Reply:
@arogowie-intel custom function was introduced because there was no NCWH format enum in oneDNN so we needed to do reorder ourselves. This is outdated as after we impleented custom reorder relevant enum was added. So actual task is to remove custom reorder and there is a issue for that in our tracker.